.github/workflows/1_pre-Test-Model-Image-Build.yaml

```diff
@@ -0,0 +1,88 @@
+name: Test Model Image Build
+run-name: (2_pre) Test Model Image - "${{ github.event.head_commit.message }}"
+
+# This GitHub Actions workflow builds a Docker image for the
+# cfa-epinow2-pipeline-docker project. In-container tests can be added here.
+
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+  schedule:
+    # Tuesdays at noon GMT time, 24 hours before the prod container is built and deployed
+    - cron: "0 12 * * 2"
+
+env:
+  # Together, these form: cfaprdbatchcr.azurecr.io/cfa-epinow2-pipeline
+  REGISTRY: cfaprdbatchcr.azurecr.io
+  IMAGE_NAME: cfa-epinow2-pipeline
+
+jobs:
+
+  Job01-build_image_dependencies:
+    runs-on: cfa-cdcgov # VM based runner serving CFA's cdcgov repos (as opposed to cdcent)
+
+    outputs:
+      tag: ${{ steps.image-tag.outputs.tag }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Figure out tag (latest if push to main, otherwise pr-<number>)
+        id: image-tag
+        run: |
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            echo "tag=pr-${{ github.event.number }}" >> $GITHUB_OUTPUT
+          else
+            echo "tag=latest" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Check cache
+        uses: actions/cache@v4
+        id: cache
+        with:
+          key: docker-dependencies-${{ runner.os }}-${{ hashFiles('./DESCRIPTION', './Dockerfile-dependencies') }}-${{ steps.image-tag.outputs.tag }}
+          lookup-only: true
+          path:
+            ./DESCRIPTION
+
+      - name: Login to the Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: "cfaprdbatchcr.azurecr.io"
+          username: "cfaprdbatchcr"
+          password: ${{ secrets.CFAPRDBATCHCR_REGISTRY_PASSWORD }}
+
+      - name: Build and push
+        if: steps.cache.outputs.cache-hit != 'true'
```

Comment:
Can we push this if block any higher in the workflow? I'd think we would want to order things to do the cache check first and then only do the rest if needed? This gets a little trickier though because of the second stage which needs some of the things computed here.....

Reply:
Yes, I'll add it to the step prior so we only log in if needed. Is that what you meant?